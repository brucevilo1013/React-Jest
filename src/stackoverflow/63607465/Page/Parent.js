import React from 'react';
import Child from './Children';

export default class Parent extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return <Child />;
  }
}
