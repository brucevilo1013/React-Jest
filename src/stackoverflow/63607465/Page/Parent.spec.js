import React from 'react';
import Parent from './Parent';

jest.mock('./Children', () => {
  return jest.fn(() => <div>mocked child</div>);
});

describe('<Parent/>', () => {
  let wrapper;
  let props = {};

  describe('render', () => {
    it('renders', () => {
      wrapper = mount(<Parent props={props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
