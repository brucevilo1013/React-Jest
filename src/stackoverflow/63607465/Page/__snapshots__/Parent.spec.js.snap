// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Parent/> render renders 1`] = `
<Parent
  props={Object {}}
>
  <mockConstructor>
    <div>
      mocked child
    </div>
  </mockConstructor>
</Parent>
`;
